core/domain/user_query_services.py

```diff
@@ -135,38 +132,21 @@ def _save_user_query(user_query):
     return user_query_model.id
 
 
-def save_new_user_query(
-        submitter_id, inactive_in_last_n_days=None,
-        has_not_logged_in_for_n_days=None, created_at_least_n_exps=None,
-        created_fewer_than_n_exps=None, edited_at_least_n_exps=None,
-        edited_fewer_than_n_exps=None):
+def save_new_user_query(submitter_id, query_params):
     """Saves a new user query.
 
     Args:
         submitter_id: str. ID of the UserQueryModel instance.
-        inactive_in_last_n_days: int. Number of days user is inactive.
-        has_not_logged_in_for_n_days: int. Number of days user hasn't logged in.
-        created_at_least_n_exps: int. Minimum number of explorations created
-            by user.
-        created_fewer_than_n_exps: int. Maximum number of explorations created
-            by user.
-        edited_at_least_n_exps: int|None. Minimum number of
-            explorations edited by user.
-        edited_fewer_than_n_exps: int|None. Maximum number of
-            explorations edited by user.
+        query_params: dict. Parameters of the USER_QUERY_PARAMS collection.
 
     Returns:
         str. The ID of the newly saved user query.
     """
+    for predicate in constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION:
+        if predicate['backend_attr'] not in query_params.keys():
```

Comment:
Should we maybe instead have tests to verify that the constants in constants.ts are properly & well-defined (e.g. that they point to valid backend methods) to avoid discovering this at runtime?

Reply:
I have added a simple test to check the existence of a function in the backend that is mentioned in the constants.